src/NServiceBus.Core.Tests/Recoverability/SecondLevelRetries/SecondLevelRetriesTests.cs

```diff
@@ -16,126 +16,158 @@
+            var context = CreateContext(message);
+            
+            var behavior = new SecondLevelRetriesBehavior(new FakePolicy(TimeSpan.FromSeconds(5)), string.Empty, failureInfoStorage);
+            var calledTwice = false;
+
+
+            await behavior.Invoke(context, () => { throw new Exception(); });
+            await behavior.Invoke(context, () =>
+            {
+                calledTwice = true;
+                return Task.FromResult(0);
+            });
+
+            Assert.IsFalse(calledTwice, "SLR should not call pipline continuation when processing a message marked for defferal.");
+        }
 
-            var message = context.Message;
-            message.Body = Encoding.UTF8.GetBytes("modified content");
+        IncomingMessage CreateMessage(string id = "id", string slrRetryHeader = null, byte[] body = null)
+        {
+            var headers = string.IsNullOrEmpty(slrRetryHeader)
+                ? new Dictionary<string, string>()
+                : new Dictionary<string, string> { { Headers.Retries, slrRetryHeader }};
 
-            await behavior.Invoke(context, () => { throw new Exception("test"); });
+            return new IncomingMessage(id, headers, new MemoryStream(body ?? new byte[0]));
+        }
 
-            var dispatchedMessage = fakeDispatchPipeline.RoutingContext.Message;
-            Assert.AreEqual(originalContent, Encoding.UTF8.GetString(dispatchedMessage.Body));
-            Assert.AreEqual(originalContent, Encoding.UTF8.GetString(message.Body));
+        FakeTransportReceiveContext CreateContext()
+        {
+            return CreateContext(CreateMessage());
         }
 
-        FakeTransportReceiveContext CreateContext(string messageId, int currentRetryCount, FakeDispatchPipeline pipeline, byte[] messageBody = null)
+        FakeTransportReceiveContext CreateContext(IncomingMessage message)
         {
-            var context = new FakeTransportReceiveContext(messageId, currentRetryCount, messageBody);
+            var context = new FakeTransportReceiveContext(message);
 
-            context.Extensions.Set<IPipelineCache>(new FakePipelineCache(pipeline));
+            context.Extensions.Set<IPipelineCache>(new FakePipelineCache(dispatchPipline));
 
             return context;
         }
 
         class FakeTransportReceiveContext : FakeBehaviorContext, ITransportReceiveContext
         {
-            public FakeTransportReceiveContext(string messageId, int currentRetryCount, byte[] messageBody)
+            public FakeTransportReceiveContext(IncomingMessage message)
             {
-                Message = new IncomingMessage(messageId, new Dictionary<string, string>
-                {
-                    {Headers.Retries, currentRetryCount.ToString()}
-                }, new MemoryStream(messageBody ?? new byte[0]));
+                Message = message;
             }
 
             public bool ReceiveOperationWasAborted { get; private set; }
```

Comment:
I don't see any testcase testing that the receive operation was aborted. Seems pretty crucial to me (since that's essentially what caused the bug), or am I missing something?